frontend/src/machines/billingMachine.ts

```diff
@@ -1,3 +1,137 @@
-import { createMachine } from "xstate";
+import { createMachine, assign } from "xstate";
 
-export const billingMachine = createMachine({});
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+interface Context {
+  billings: Billing[];
+  isEditingForm: boolean;
+  isCreatingForm: boolean;
+}
+
+const billingMachine = createMachine<Context>({
+  id: "billing",
+  initial: "idle",
+  context: {
+    billings: [],
+    isCreatingForm: false,
+    isEditingForm: false,
+  },
+  states: {
+    idle: {
+      on: {
+        FETCHINGBILLINGS: {
```

Comment:
Kayaknya ini perlu dibuat snake case biar mudah dibaca ?

```suggestion
        FETCHING_BILLINGS: {
```

Reply:
Hmmm bener juga, aku seragamkan aja yak.
Buat Event aku jadikan snake case aja